### DISCUSSION TITLE
Supporting Cassandra/Yugabyte as airflow backend DB

### DISCUSSION THREAD

**Discussion by tetsushiawano** [apache/airflow#32675] [General]
### Description

Wishing compatability for active-active database as backend DB of airflow.

The current supported DB are not active-active DB by nature in my understanding. (or expensive to enable active-active feature.)

Is there any plan to support database like Cassandra or YugabyteCQL in future?

### Use case/motivation

Make multi-region active-active Airflow hosted on our multiple OnPrem Datacenter.

<img width="1080" alt="Screenshot 2023-07-18 at 14 18 07" src="https://github.com/apache/airflow/assets/24989207/aa7a0501-37b1-44cc-93ac-23228df340fd">

### Are you willing to submit a PR?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/apache/airflow/blob/main/CODE_OF_CONDUCT.md)

**Comment by boring-cyborg** [comment]
Thanks for opening your first issue here! Be sure to follow the issue template! If you are willing to raise PR to address this issue please do so, no need to wait for approval.

**Comment by raphaelauv** [comment]
Airflow expect a relational database because it use SqlAlchemy

for Yugabyte , maybe you could make it work , but few postgres "features" are missing -> https://github.com/yugabyte/yugabyte-db/issues/5683

**Comment by nathadfield** [comment]
@potiuk Would you like to comment on this from the perspective of project direction and priorities?

**Comment by potiuk** [comment]
Not possible without completely rewriting hiw scheduler works. We heavily rely on lock mechanisms in relation databases. That does not work well with active/active setup

And it is not needed IMHO.

Generally speaking Airflow is not targeting  99.9% available system(8 hours a year). It is far less availability as target. And this is by design. Airflow should not be used as a backbone of system that requires it. It is a batch scheduling system. Having more availability for Airflow is NON-goal for us (at least so far it was not).

Every single 9 there order of magnitude complexity and cost of development and requires usually much more complex maintrnance. If someone wills to invest 10x the time that has been used to develop current scheduler - maybe they should. But we never saw it as a goal. That's why we rejected things like zookeeper, non-relational databases and supporting active-active setup in the past 

Supporting such setup without real need is at most following a 'trend' without really considering if it is worth the cost. You have to remember that every single decision of architecture bears a cost. And in this case - the cost is huge.

If someone would like to spend time and write Airflow Improvement Proposal and convince others and bear the cost of implementing it - yes sure, anyone is free to start it. But considering cost of implementation, deployment and deeply considering what it means to run Airflow in such a way, reasoning why it is necessary will be a crucial part of it. And convincing community members that it is worth it is the first step.

For now I personally am not convinced. But luckily it's not me to convince. Bring it to devlist, state your reasoning, fend off people who will not like it, reach consensus, get the AIP to pass the vote  and implement it. This is the way how one should approach it.

**Comment by potiuk** [comment]
Converted it to discussion, because this is really what it is. Any more comments are welcome, but any proposals here will have to go to devlist, and AIP has to be prepared.

**Comment by tetsushiawano** [comment]
mr @potiuk mr @raphaelauv 
Thank you for the very details feedback.
I plan use these valuable feedback to consider for our next step which were focusing on. 
Thank you so much again.
